fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMinMaxAggToMetaScanRule.java

```diff
@@ -0,0 +1,162 @@
+        LogicalScanOperator scanOperator = (LogicalScanOperator) optExpression.getOp();
+        OlapTable table = (OlapTable) scanOperator.getTable();
+        // we can only apply this rule to the queries met all the following conditions:
+        // 1. query on DUPLICATE_KEY table
+        // 2. no group by key
+        // 3. no `having` condition or other filters
+        // 4. no limit
+        // 5. only contain MIN/MAX agg functions
+        // 6. all arguments to agg functions are primitive columns
+        // 7. no expr in arguments to agg functions
+        // 8. all agg columns have zonemap index
+        if (table.getKeysType() != KeysType.DUP_KEYS) {
+            return new ArrayList<>();
+        }
+        List<ColumnRefOperator> groupingKeys = aggregationOperator.getGroupingKeys();
+        if (groupingKeys != null && !groupingKeys.isEmpty()) {
+            return new ArrayList<>();
+        }
+        if (aggregationOperator.getPredicate() != null) {
+            return new ArrayList<>();
+        }
+        boolean isOnlySimpleMinMax = aggregationOperator.getAggregations().values().stream().allMatch(
+                aggregator -> {
+                    AggregateFunction aggregateFunction = (AggregateFunction) aggregator.getFunction();
+                    String functionName = aggregateFunction.functionName();
+                    if (!(functionName.equals(FunctionSet.MAX) || functionName.equals(FunctionSet.MIN))) {
+                        return false;
+                    }
+                    ColumnRefSet usedColumns = aggregator.getUsedColumns();
+                    if (usedColumns.size() != 1) {
+                        return false;
+                    }
+                    ColumnRefOperator usedColumn = context.getColumnRefFactory().getColumnRef(usedColumns.getFirstId());
+                    if (scanOperator.getColRefToColumnMetaMap().get(usedColumn) == null) {
+                        // this is not a primitive column on table
+                        return false;
+                    }
+                    return true;
+                }
+        );
+        if (!isOnlySimpleMinMax) {
+            return new ArrayList<>();
+        }
+        boolean allCanUseZonemapIndex = aggregationOperator.getAggregations().values().stream().allMatch(
+                aggregator -> {
+                    Type type = aggregator.getType();
+                    return !(type.isStringType() || type.isComplexType());
+                }
+        );
+        // all agg columns have zonemap index
+        if (!allCanUseZonemapIndex) {
+            return new ArrayList<>();
+        }
+        // no limit
+        if (scanOperator.getLimit() != -1) {
+            return new ArrayList<>();
+        }
+        Map<String, PartitionColumnFilter> columnFilters = scanOperator.getColumnFilters();
+        // no filter
+        if (columnFilters != null && !columnFilters.isEmpty()) {
+            return new ArrayList<>();
+        }
```

Comment:
check scan's predicate, not getColumnFilters

Reply:
btw, metaScanNode doesn't support work on special tablet/partitions?

Reply:
> check scan's predicate

fixed

Reply:
> btw, metaScanNode doesn't support work on special tablet/partitions?

it supports…I made a mistake here before